File: tests/benchmark/corpus/python/safe/safe_optional_path_sanitizer.py
# py-safe-015: Optional[str]-returning sanitiser with None failure sentinel.
import os
from typing import Optional
from flask import request


def sanitize_path(s: str) -> Optional[str]:
    if ".." in s or s.startswith("/") or s.startswith("\\"):
        return None
    return s


def main() -> None:
    raw = request.args.get("path")
    safe = sanitize_path(raw)
    if safe is None:
        return
    with open(safe) as f:
        f.read()

File: tests/benchmark/corpus/java/safe/SafeOptionalPathSanitizer.java
// java-safe-015: Optional<String>-returning sanitiser with `.empty()` failure sentinel.
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Optional;

public class SafeOptionalPathSanitizer {
    public static Optional<String> sanitizePath(String s) {
        if (s.contains("..") || s.startsWith("/") || s.startsWith("\\")) {
            return Optional.empty();
        }
        return Optional.of(s);
    }

    public static void handle(String userPath) throws IOException {
        Optional<String> safe = sanitizePath(userPath);
        if (!safe.isPresent()) return;
        Files.readAllBytes(Paths.get(safe.get()));
    }
}
